04-onion-routing.md

```diff
@@ -853,7 +853,8 @@ The top byte of `failure_code` can be read as a set of flags:
 * 0x1000 (UPDATE): new channel update enclosed
 
 Please note that the `channel_update` field is mandatory in messages whose
-`failure_code` includes the `UPDATE` flag.
+`failure_code` includes the `UPDATE` flag. It is encoded *with* the message
```

Comment:
I think we need to more explicitly spell out backwards compatibility here: nodes _should_ start to write with the type prefix, but also continue to read _both_ types. Otherwise, we'd essentially fracture the network as the new cohort of nodes will only assume/read the type byte instead of always try to read both.

Reply:
Agreed with Matt on the forward-looking philosophy, but we can add just a comment saying that it used to be encoded without the type in some earlier versions if roasbeef finds it helpful?

Reply:
Outcome of the meeting is that a small comment here could be helpful, and with it we should be good to go :+1: